### PR TITLE
NAS-113588 / 22.02 / NAS-113588: Allow empty string as an FTP password

### PR DESCRIPTION
**Pull request by undsoft**
Credentials -> Backup credentials
Empty FTP password + regressions.

### CODE REVIEW

File: src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts [hunk 1]
```diff
@@ -1244,26 +1244,8 @@ export class CloudCredentialsFormComponent implements FormConfiguration {
       function: () => {
         this.entityForm.loader.open();
         this.entityForm.error = '';
-        const attributes: any = {};
         const value = _.cloneDeep(this.entityForm.formGroup.value);
-        this.beforeSubmit(value);
```
Comment (AlexKarpov98) [inline src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts:1249]: Please don't skip the call `beforeSubmit` function.
If OAuth URL is null, should delete value `client_id` and `client_secret`.

File: src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts [hunk 2]
```diff
@@ -1244,26 +1244,8 @@ export class CloudCredentialsFormComponent implements FormConfiguration {
...
-        let attrName: string;
-
-        for (const item in value) {
-          if (item != 'name' && item != 'provider') {
-            if (!this.entityForm.formGroup.controls[item].valid) {
-              this.entityForm.formGroup.controls[item].markAsTouched();
-            }
```
Comment (AlexKarpov98) [inline src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts:1256]: Same as above. don't skip for triggering validation

**Review by AlexKarpov98 — CHANGES_REQUESTED** [review]
Looks good!

File: src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts (hunk 1 above)
Comment (undsoft) [reply src/app/pages/credentials/backup-credentials/forms/cloud-credentials-form.component.ts]: Weird that I removed it. Fixed.

**Review by AlexKarpov98 — APPROVED** [review]
Great!

**Pull request merged**